datafusion/core/src/physical_plan/planner.rs

```diff
@@ -292,10 +293,10 @@ pub trait ExtensionPlanner {
     /// Returns `None` when the planner does not know how to plan the
     /// `node` and wants to delegate the planning to another
     /// [`ExtensionPlanner`].
-    fn plan_extension(
+    async fn plan_extension(
         &self,
-        planner: &dyn PhysicalPlanner,
-        node: &dyn UserDefinedLogicalNode,
+        planner: &(dyn PhysicalPlanner + Send + Sync),
```

Comment:
I wonder if the `Send` and `Sync` constraints should be added to `PhysicalPlanner` and `UserDefinedLogicalNode` instead of at this specific call-site?

Reply:
Done. I also add `Send` and `Sync` bounds to `UserDefinedLogicalNode`.

I think these are also "API change"?